test/e2e/internal/utils/registry.go

```diff
@@ -0,0 +1,28 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import "github.com/opencontainers/go-digest"
+
+// Reference generates the reference string from given parameters.
+func Reference(registry string, repo string, tagOrDigest string) string {
+	ret := registry + "/" + repo
+
+	if err := digest.Digest(tagOrDigest).Validate(); err != nil {
+		ret += ":" + tagOrDigest
+	} else {
+		ret += "@" + tagOrDigest
+	}
+	return ret
```

Comment:
We probably need to document the reason why we don't use `registry.Reference` from` oras-go` to generate the reference string.

Reply:
Actually we should, why not.